src/main/java/omero/gateway/ServerInformation.java

```diff
@@ -30,57 +33,139 @@
 
 public class ServerInformation {
 
-    /** The hostname */
-    private String hostname;
-
-    /** The port */
-    private int port;
+    /** The URI */
+    private URI uri;
 
     /**
      * Creates an empty instance
      */
     public ServerInformation() {
+        try {
+            this.uri = new URI(null, null, null,
+                    -1, null, null, null);
+        } catch (URISyntaxException e) {
+        }
+    }
 
+    /**
+     * Creates a new instance
+     *
+     * @param hostname
+     *            The hostname or websocket URL
+     */
+    public ServerInformation(String hostname) {
+        this(hostname, -1);
     }
 
     /**
      * Creates a new instance
      * 
      * @param hostname
-     *            The hostname
+     *            The hostname or websocket URL
      * @param port
      *            The port
      */
     public ServerInformation(String hostname, int port) {
-        super();
-        this.hostname = hostname;
-        this.port = port;
+        try {
+            if (hostname.contains(":/")) {
+                // this is already a URI like wss://example.org
+                this.uri = new URI(hostname);
+                if (port >= 0 && this.uri.getPort() < 0) {
+                    this.uri = new URI(uri.getScheme(), uri.getUserInfo(), uri.getHost(),
+                            port, uri.getPath(), uri.getQuery(), uri.getFragment());
+                }
+            }
+            else {
+                if (port < 0 && hostname.contains(":")) {
+                    try {
+                        String[] parts = hostname.split(":");
+                        port = Integer.parseInt(parts[parts.length-1]);
+                        hostname = parts[parts.length-2];
+                    } catch (Exception e) {}
+                }
+                this.uri = new URI(null,null, hostname,
+                        port, null, null, null);
+            }
+        } catch (URISyntaxException e) {
```

Comment:
Do you want to log the error?

Reply:
Thought about it. But at that point I don't have any reference to the Gateway, which has the Logger.

Reply:
ok understood